utils/zerovec/src/zerovec/mod.rs

```diff
@@ -554,6 +555,34 @@ where
             Some(ZeroSlice::from_ule_slice(ule_slice))
         }
     }
+
+    /// If the ZeroVec is owned, returns the capacity of the vector.
+    ///
+    /// Otherwise, if the ZeroVec is borrowed, returns `None`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use zerovec::ZeroVec;
+    ///
+    /// let mut zv = ZeroVec::<u8>::new_borrowed(&[0, 1, 2, 3]);
+    /// assert!(!zv.is_owned());
+    /// assert_eq!(zv.owned_capacity(), None);
+    ///
+    /// // Convert to owned without appending anything
+    /// zv.with_mut(|v| ());
+    /// assert!(zv.is_owned());
+    /// assert_eq!(zv.owned_capacity(), Some(4.try_into().unwrap()));
+    ///
+    /// // Double the size by appending
+    /// zv.with_mut(|v| v.push(0));
+    /// assert!(zv.is_owned());
+    /// assert_eq!(zv.owned_capacity(), Some(8.try_into().unwrap()));
+    /// ```
+    #[inline]
+    pub fn owned_capacity(&self) -> Option<NonZeroUsize> {
```

Comment:
thought: I think having the method be  `.capacity()` and returning an integer is also fine, "Borrowed is zero capacity" is a consistent and sensical statement, and generally useful. (if most people are going to `unwrap_or(0)` this API, might as well)


but not opposed to this

Reply:
I wanted to be more type-safe and explicit.